zebra-consensus/src/chain/tests.rs

```diff
@@ -426,20 +431,63 @@ async fn continuous_blockchain() -> Result<(), Report> {
         blockchain.push((block.clone(), block.coinbase_height().unwrap(), hash));
     }
 
-    // Make a checkpoint list containing the genesis block
-    let checkpoint_list: BTreeMap<BlockHeight, BlockHeaderHash> = blockchain
+    // Parse only some blocks as checkpoints
+    let mut checkpoints = Vec::new();
+    for b in &[
+        &zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..],
+        &zebra_test::vectors::BLOCK_MAINNET_4_BYTES[..],
+    ] {
+        let block = Arc::<Block>::zcash_deserialize(*b)?;
+        let hash: BlockHeaderHash = block.as_ref().into();
+        checkpoints.push((block.clone(), block.coinbase_height().unwrap(), hash));
+    }
+
+    // The checkpoint list will contain only blocks 0 and 4
+    let checkpoint_list: BTreeMap<BlockHeight, BlockHeaderHash> = checkpoints
         .iter()
-        .map(|(_, height, hash)| (*height, *hash))
-        .take(1)
+        .map(|(_block, height, hash)| (*height, *hash))
         .collect();
     let checkpoint_list = CheckpointList::from_list(checkpoint_list).map_err(|e| eyre!(e))?;
 
-    let (mut chain_verifier, _) = verifiers_from_checkpoint_list(Mainnet, checkpoint_list);
+    let mut state_service = zebra_state::in_memory::init();
+    /// SPANDOC: Add blocks from 0..={?restart_height} to the state
```

Comment:
spandoc only parses `{}` blocks at the very end of the output, spans don't support interpolation unlike tracing events, the body of the comment is just used as a `text` field in the span, and every field in the block at the end is then added as additional fields to capture.

I could probably emulate interpolation in this case though, it would generate a span that looks like this, lmk what you think.

```rust
span!(Level::ERROR, "fn_name::comment", text = "Add blocks from 0..=restart_height to the state", ?restart_height);
```

Reply:
I think emulating interpolation would be great (and much less surprising than requiring a block at the end)